detection_rules/rule.py

```diff
@@ -395,6 +401,46 @@ def setup_in_note(self, value: bool):
     def skip_validate_note(self) -> bool:
         return os.environ.get('DR_BYPASS_NOTE_VALIDATION_AND_PARSE') is not None
 
+    @cached_property
+    def skip_validate_bbr(self) -> bool:
+        return os.environ.get('DR_BYPASS_BBR_LOOKBACK_VALIDATION') is not None
+
+    def validate_bbr(self):
+        if self.skip_validate_bbr:
+            return
+
+        def validate_time_defaults(str_time) -> bool:
+            """Validate that the time is at least now-119m and at least 60m respectively."""
+            try:
+                if "now" in str_time:
+                    str_time = str_time.replace("now-", "")
+                    time = convert_time_span(str_time)
+                    # if time is less than 119m as milliseconds
+                    if time < 119 * 60 * 1000:
+                        return False
+                else:
+                    time = convert_time_span(str_time)
+                    # if time is less than 60m as milliseconds
+                    if time < 60 * 60 * 1000:
+                        return False
+            except Exception as e:
+                raise ValidationError(f"Invalid time format: {e}")
+            return True
+
+        """Validate building block type and rule type."""
+        if self.building_block_type:
+            if not self.from_ or not self.interval:
+                raise ValidationError(
+                    "BBR require `from` and `interval` to be defined. Please set or bypass."
+                    "To bypass, use the environment variable `DR_BYPASS_BBR_LOOKBACK_VALIDATION`"
```

Comment:
You could assign this to a var and use it here and anywhere below.

Reply:
updated